device.go

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/kata-containers/agent/pkg/uevent"
+	pb "github.com/kata-containers/agent/protocols/grpc"
+	"github.com/sirupsen/logrus"
+)
+
+type deviceDriversHandler func(device pb.Storage, spec *pb.Spec) error
+
+var deviceDriversHandlerList = map[string]deviceDriversHandler{
+	"blk": blockDeviceHandler,
+}
+
+func blockDeviceHandler(device pb.Storage, spec *pb.Spec) error {
+	if spec.Linux == nil || len(spec.Linux.Devices) == 0 {
+		return fmt.Errorf("No devices found from the spec, cannot update")
+	}
+
+	// First need to make sure the expected device shows up properly,
+	// and then we need to retrieve its device info (such as major and
+	// minor numbers), useful to update the device provided
+	// through the OCI specification.
+	if err := waitForDevice(device.Source); err != nil {
+		return err
+	}
+
+	stat := syscall.Stat_t{}
+	if err := syscall.Stat(device.Source, &stat); err != nil {
+		return err
+	}
+
+	major := int64(stat.Rdev / 256)
+	minor := int64(stat.Rdev % 256)
+
+	agentLog.Infof("Device: %s, Major: %d, Minor: %d", device.Source, major, minor)
+
+	// Update the spec
+	updated := false
+	for idx, d := range spec.Linux.Devices {
+		if d.Path == device.MountPoint {
+			spec.Linux.Devices[idx].Major = major
+			spec.Linux.Devices[idx].Minor = minor
+			updated = true
+			break
+		}
+	}
+
+	if !updated {
+		return fmt.Errorf("Should have found a matching device in the spec")
```

Comment:
I guess there is the previous log call, but it might be useful to atleast log `device.Source` here so it's clear which device wasn't found in the spec.

Reply:
done